pydantic_ai_slim/pydantic_ai/models/mistral.py

```diff
@@ -0,0 +1,575 @@
+            content = choice.message.content
+            assert content, f'Unexpected null content is assitant msg: {choice.message}'
+            assert not isinstance(
+                content, list
+            ), f'Unexpected ContentChunk from stream, need to be response not stream: {content}'
+            return ModelTextResponse(content, timestamp=timestamp)
+
+    @staticmethod
+    async def _process_streamed_response(
+        is_function_tools: bool,
+        result_tools: list[ToolDefinition] | None,
+        response: MistralEventStreamAsync[MistralCompletionEvent],
+    ) -> EitherStreamedResponse:
+        """Process a streamed response, and prepare a streaming response to return."""
+        start_cost = Cost()
+
+        # Iterate until we get either `tool_calls` or `content` from the first chunk.
+        while True:
+            try:
+                event = await response.__anext__()
+                chunk = event.data
+            except StopAsyncIteration as e:
+                raise UnexpectedModelBehavior('Streamed response ended without content or tool calls') from e
+
+            start_cost += _map_cost(chunk)
+
+            timestamp: datetime
+            if chunk.created:
+                timestamp = datetime.fromtimestamp(chunk.created, tz=timezone.utc)
+            else:
+                timestamp = _now_utc()
+
+            if chunk.choices:
+                delta = chunk.choices[0].delta
+                content = _map_delta_content(delta.content)
+
+                tool_calls: list[MistralToolCall] | None = None
+                if isinstance(delta.tool_calls, list):
+                    tool_calls = delta.tool_calls
+
+                if content and result_tools:
+                    return MistralStreamStructuredResponse(
+                        is_function_tools,
+                        result_tools,
+                        response,
+                        None,
+                        content,
+                        timestamp,
+                        start_cost,
+                    )
+
+                elif content:
+                    return MistralStreamTextResponse(content, response, timestamp, start_cost)
+
+                elif tool_calls and not result_tools:
+                    tool_calls_param = {c.id if c.id else 'null': c for c in tool_calls}
+                    return MistralStreamStructuredResponse(
+                        is_function_tools,
+                        result_tools,
+                        response,
+                        tool_calls_param,
+                        content,
+                        timestamp,
+                        start_cost,
+                    )
+
+    @staticmethod
+    def _map_message(message: Message) -> MistralMessages:
+        """Just maps a `pydantic_ai.Message` to a `Mistral Message`."""
+        if message.role == 'system':
+            # SystemPrompt ->
+            return MistralAssistantMessage(content=message.content)
```

Comment:
this looks wrong, it shouldn't be an assistant message, if theres no specific system prompt type, I would use `MistralUserMessage`.

Unless I'm missing something?